### PR TITLE
Feature: LEAF-2441 - expand report builder for grids

### PR DESCRIPTION
**Pull request by raphaelfretter**

### CODE REVIEW

**Review by AmandaSumner — APPROVED** [review]

**Review by maxrichardjr — APPROVED** [review]
Looks good.

**Review by Pelentan — APPROVED** [review]
Looks good.

File: LEAF_Request_Portal/js/formGrid.js [hunk 1]
```diff
@@ -45,21 +45,30 @@ var LeafFormGrid = function(containerID, options) {
...
+            return columnValues.includes(value.id);
+        });
+
+
         //finds and displays column names
         for(var i = 0; i < columns; i++){
-            tDelim = (i == columns-1) ? '' : delim;
-            gridHeadBuffer +='<td style="width: 100px;">' + values.format[i].name + tDelim + '</td>';
-            columnOrder.push(values.format[i].id)
+            if (columnValues.includes(values.format[i].id)) {
```
Comment (pete-nerantzinis) [inline LEAF_Request_Portal/js/formGrid.js:67]: Getting error here during testing, 
image

**Review by pete-nerantzinis — CHANGES_REQUESTED** [review]
Found some errors during testing that caused the report builder to crash. Line 67 of formGrid.js and a few others I could not determine.

File: LEAF_Request_Portal/js/formGrid.js (hunk 1 above)
Comment (aerinkayne) [reply LEAF_Request_Portal/js/formGrid.js:67]: I got the same.  I sent him an image of the checkbox conditions that seemed to be associated with it.

Comment (pete-nerantzinis) [reply LEAF_Request_Portal/js/formGrid.js]: This looks to be working now

**Review by pete-nerantzinis — CHANGES_REQUESTED** [review]
I was just about to approve but then came across this issue.

image

Seems when you create a report then refresh the page to get the cached version of the report from sharing the url, it does not save the grid selections. It will refresh properly at first but if you go to modify and regenerate it will give the error above, and closer looking on modify on step 2 you can see only the main checkbox is checked and not the sub checkboxes.

To Rrecreate:
1. Go to Report builder and setup report for something that shows grids checkbox the main grid so it selects all grid items
2. Generate Report
3. Refresh the page so its on the cached version where you cannot edit the main title
4. Modify again and go to step 2, notice that the checkboxes for the sub grid items are not checked anymore, then when generate report it bugs out

File: LEAF_Request_Portal/templates/view_reports.tpl [hunk 2]
```diff
@@ -462,6 +541,13 @@ function loadSearchPrereqs() {
                             if(t_inIndicators != undefined) {
                                 for(var i in t_inIndicators) {
                                     $('#indicators_' + t_inIndicators[i].indicatorID).prop('checked', true);
+
+                                    if (t_inIndicators[i].cols !== undefined) {
+                                        for (var j in t_inIndicators[i].cols) {
+                                            console.log('#indicators_' + t_inIndicators[i].indicatorID + '_columns_' + t_inIndicators[i].cols[j]);
```
Comment (pete-nerantzinis) [inline LEAF_Request_Portal/templates/view_reports.tpl:547]: So it mostly looks good now, just clean up this last console log unless its necessary for users. 

Other than that, I was doing some backwards compatibility with reports created before this feature was implemented with a grid field type. It doesn't work it gives me this error:
image

Just make sure its also backwards compatible by moving back to master branch and creating a report with grids and saving the link then checking out the new branch and testing that link for backwards compatibility.

I think this is good to go after that is done.

**Review by pete-nerantzinis — CHANGES_REQUESTED** [review]
Sorry Raphael, I caught one last thing!

**Review by beingforthebenefit — APPROVED** [review]

File: LEAF_Request_Portal/js/formGrid.js [hunk 3]
```diff
@@ -45,7 +45,14 @@ var LeafFormGrid = function(containerID, options) {
...
      */
-    function printTableReportBuilder(values) {
+    function printTableReportBuilder(values, columnValues) {
+        // remove unused columns
+        if (columnValues !== null) {
+            values.format = values.format.filter(function (value) {
+                return columnValues.includes(value.id);
+            });
+        }
+
```
Comment (aerinkayne) [inline LEAF_Request_Portal/js/formGrid.js:55]: I get an error here if I try to edit/save the grid section.  Adding && !== undefined will fix it.  After editing, the grid resets to the full grid (I don't know if that's an issue or not.  The dialog boxes are their own little monsters).

**Review by aerinkayne — APPROVED** [review]

**Pull request merged**